File: codesignal/arcade/Intro/JS/reverseInParentheses.js
function reverseInParentheses(inputString) {
    let positions = []
    let stack = []
    for(let i = 0; i < inputString.length; i++){
        if(inputString[i] == "("){ stack.push(i) }
        if(inputString[i] == ")"){ 
            let pos = stack.pop()
            let value = inputString.substring(pos, i).split("").reverse().join("")
            inputString = inputString.substring(0, pos) + value + inputString.substring(i, inputString.length)
        }
    }
    
    return inputString.replace(/[\(\)]/g, "")
}

File: codesignal/arcade/Intro/Csharp/reverseInParentheses.cs
string resverseInParentheses(string inputString){
    Stack<int> stack = new Stack<int>();

    for(int i = 0; i < str.Length; i++){
        if(str[i] == '(') stack.Push(i);
        if(str[i] == ')'){
            int pos = stack.Pop();
            string text = str.Substring(pos+1, i - pos - 1);
            char[] charsText = text.ToCharArray();
            Array.Reverse(charsText);
            string reversedString = string.Join("", charsText);
            str = str.Substring(0, pos - 0 + 1) + reversedString + str.Substring(i, str.Length - i);
        }
    }
    
    return Regex.Replace(str, @"[\(\)]", "");
}